### PR TITLE
[12.0][FIX] partner recursion check

### PR DESCRIPTION
**Pull request by sergiocorato**
Current behavior: it is possible to select the partner itself as its company group
Desired behavior: recursion is not allowed

### CODE REVIEW

File: partner_company_group/models/res_partner.py [hunk 1]
```diff
@@ -13,5 +14,11 @@ class Contact(models.Model):
...
     )
 
+    @api.constrains('company_group_id')
+    def _check_company_group_id(self):
+        if not self._check_recursion('company_group_id'):
+            raise ValidationError(
+                _('Error! You cannot create recursive partner company group'))
+
     def _commercial_fields(self):
         return super(Contact, self)._commercial_fields() + ['company_group_id']
```
Comment (NL66278) [inline partner_company_group/models/res_partner.py]: Just use super(), this is python 3.x, repeating the class name with self is no longer needed.

**Review by NL66278 — APPROVED** [review]
:+1: LGTM. You can either take up my other comment, or ignore it, as it is outside of the scope of what you want to fix.

File: partner_company_group/models/res_partner.py (hunk 1 above)
Comment (sergiocorato) [reply partner_company_group/models/res_partner.py]: Ook, I was not changing that part, but I will do, thanks

**Comment by sergiocorato** [comment]
> +1 LGTM. You can either take up my other comment, or ignore it, as it is outside of the scope of what you want to fix.

+1

**Pull request merged**